sentry/src/test/java/io/sentry/SentryTracerTest.kt

```diff
@@ -0,0 +1,283 @@
+package io.sentry
+
+import com.nhaarman.mockitokotlin2.check
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.spy
+import com.nhaarman.mockitokotlin2.verify
+import io.sentry.protocol.App
+import io.sentry.protocol.Request
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNotEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+class SentryTracerTest {
+
+    private class Fixture {
+        val hub: Hub
+
+        init {
+            val options = SentryOptions()
+            options.dsn = "https://key@sentry.io/proj"
+            hub = spy(Hub(options))
```

Comment:
just out of curiosity, what would be the benefit of using `spy` instead of `mock` as usual? are we asserting the `hub` itself? if so, should we not test that thru the HubTest class?

Reply:
With using mock we can't really test that transaction is cleared from the scope. If we use mock for `finish` method, we won't really test much.